src/bird.cpp

```diff
@@ -0,0 +1,85 @@
+#include "bird.h"
+
+const float Bird::BIRD_SCALE = 0.2;
+const int Bird::COLLISION_TOLERANCE = 25;
+const int Bird::NUM_COLLISIONS_TO_TRIGGER_PER_SECOND = 300;
+
+Bird::Bird(sf::Vector2f center, int frameRate)
+{
+    bool res = birdTexture.loadFromFile("./lib/bird.png");
+    if(!res) throw("Cannot load bird\n");
+    bird.setTexture(birdTexture, true);
+    bird.setScale(BIRD_SCALE, BIRD_SCALE);
+
+    sf::Vector2f dimensions = bird.getLocalBounds().getSize();
+    this->bird.setOrigin(dimensions.x / 2, dimensions.y / 2);
+    this->bird.setPosition(center);
+
+    speed = 0;
+    this->frameRate = frameRate;
+
+    acc = 9.8 / frameRate * 1.5;
+    negAcc = acc * 2;
+
+    collisionCounter = 0;
+    NUM_COLLISIONS_TO_TRIGGER = NUM_COLLISIONS_TO_TRIGGER_PER_SECOND / frameRate;
+}
+
+sf::Sprite Bird::updatePosition()
+{
+    speed += (speed >= 0 ? negAcc : acc);
+
+    sf::Vector2f position = this->bird.getPosition();
+    position.y += speed;
+    bird.setPosition(position);
+
+    return bird;
+}
+
+void Bird::jump()
+{
+    speed = -7;
+}
+
+bool Bird::checkCollision(sf::Sprite *pipeSprites, int numPipes){
+    sf::Vector2f birdPos = bird.getPosition();
+    float width = bird.getGlobalBounds().width;
+    float height = bird.getGlobalBounds().height;
+    float radius = (width < height ? width : height);
+
+    bool isCollision = false;
+    
+    for(int i = 0 ; i < numPipes ; i++)
+    {
+        sf::FloatRect pipeRect = pipeSprites[i].getGlobalBounds();
+        float testX = birdPos.x;
+        float testY = birdPos.y;
+
+        if(birdPos.x < pipeRect.left) testX  = pipeRect.left;
+        else if(birdPos.x > pipeRect.left + pipeRect.width) testX = pipeRect.left + pipeRect.width;
+
+        if(birdPos.y < pipeRect.top) testY  = pipeRect.top;
+        else if(birdPos.y > pipeRect.top + pipeRect.height) testY = pipeRect.top + pipeRect.height;
+
+        float distX = birdPos.x - testX;
+        float distY = birdPos.y - testY;
+        float distance = sqrt((distX * distX) + (distY * distY)); 
+
+        if(radius - distance > COLLISION_TOLERANCE)
+        {
+            collisionCounter++;
+            isCollision = true;
+            if(collisionCounter > NUM_COLLISIONS_TO_TRIGGER) return true;
```

Comment:
I don't understand why you have a collision counter/ num collisions to trigger. Are you only GGing if hte player collides multiple times in the same frame??